packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -781,10 +783,21 @@ class IOSDevice extends Device {
       vmUrlFromMDns,
       // vmServiceDiscovery uses device logs (`idevicesyslog`), which doesn't work
       // on wireless devices.
-      if (vmServiceDiscovery != null && !isWirelesslyConnected) vmServiceDiscovery.uri,
+      if (discoverVMUrlFromLogs) vmServiceDiscovery.uri,
     ];
 
-    Uri? localUri = await Future.any(<Future<Uri?>>[...discoveryOptions, cancelCompleter.future]);
+    Uri? localUri;
+    try {
+      localUri = await Future.any(<Future<Uri?>>[...discoveryOptions, cancelCompleter.future]);
+    } on MissingLocalNetworkPermissionsException catch (e) {
+      if (discoverVMUrlFromLogs) {
+        // If mDNS fails, don't throw since url may still be findable through vmServiceDiscovery.
+        _logger.printError(e.toString());
+        localUri = await vmServiceDiscovery.uri;
+      } else {
+        rethrow;
```

Comment:
Should this be a tool exit? We don't want this exception to show up in crash analytics if the user needs to take action to fix this.

Reply:
Do all exceptions go to crash analytics? Like if this throws, it doesn't generate a crash report. @bkonyi Do you know?

Reply:
Yeah, unhandled exceptions are captured by crash analytics. If we encounter a fatal error that isn't a crash or unexpected, we should tool exit instead.

Reply:
@loic-sharma I'm going to revert back to passing down a `throwOnError` parameter so that we don't need to catch and rethrow everywhere `MDnsVmServiceDiscovery` is used

Reply:
I'm not exactly sure how this works, but my understanding is that tool exists are treated as a user error and are excluded from the tool's exception tracking.